custom_components/moonraker/sensor.py

```diff
@@ -99,48 +100,71 @@ class MoonrakerSensorDescription(SensorEntityDescription):
         subscriptions=[("print_stats", "filename")],
     ),
     MoonrakerSensorDescription(
-        key="print_projected_duration",
-        name="print Projected Duration",
+        key="print_projected_total_duration",
+        name="print Projected Total Duration",
+        value_fn=lambda data: round(
+            data["status"]["print_stats"]["print_duration"] / calculate_pct_job(data)
+            if calculate_pct_job(data) != 0
+            else 0,
+            2,
+        ),
+        subscriptions=[
+            ("print_stats", "total_duration"),
+            ("display_status", "progress"),
+        ],
+        icon="mdi:timer",
+        unit=TIME_SECONDS,
+        device_class=SensorDeviceClass.DURATION,
+    ),
+    MoonrakerSensorDescription(
+        key="print_time_left",
+        name="Print Time Left",
         value_fn=lambda data: round(
             (
-                (data["status"]["print_stats"]["print_duration"] / 60)
-                / data["status"]["display_status"]["progress"]
-            ),
+                data["status"]["print_stats"]["print_duration"]
+                / calculate_pct_job(data)
+                if calculate_pct_job(data) != 0
+                else 0
+            )
+            - data["status"]["print_stats"]["print_duration"],
             2,
-        )
-        if data["status"]["display_status"]["progress"] != 0
-        else 0,
+        ),
         subscriptions=[
             ("print_stats", "print_duration"),
             ("display_status", "progress"),
         ],
         icon="mdi:timer",
-        unit=TIME_MINUTES,
+        unit=TIME_SECONDS,
         device_class=SensorDeviceClass.DURATION,
     ),
     MoonrakerSensorDescription(
         key="print_eta",
-        name="ETA",
+        name="Print ETA",
         value_fn=lambda data: (
```

Comment:
At that point, I need those functions should be written differently for better readability and not make the code heavier. It may be my limited experience with Python, but I can't parse those one liner